### PR TITLE
Add post review response and move review count to be a response header

### PR DESCRIPTION
**Pull request by JR40159**

### CODE REVIEW

File: backend/src/services/v2/review.ts [hunk 1]
```diff
@@ -81,9 +54,80 @@ export async function createReleaseReviews(model: ModelDoc, release: ReleaseDoc)
...
   const roleEntities: string[] = collaborators
     .filter((collaborator) => collaborator.roles.includes(role))
     .map((collaborator) => collaborator.entity)
   return roleEntities
 }
+
+/**
+ * Return the models where one of the user's entities is in the model's collaberators
+ * and the role in the review is in the list of roles in that collaborator entry.
+ */
```
Comment (a3957273) [inline backend/src/services/v2/review.ts:106]: Could you add that this requires the `model` attribute?

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: backend/src/routes/v2/review/getReviews.ts [hunk 2]
```diff
@@ -21,9 +23,11 @@ export const getReviews = [
   bodyParser.json(),
   async (req: Request, res: Response<GetReviewResponse>) => {
     const {
-      query: { active, modelId },
+      query: { active },
+      params: { modelId },
     } = parse(req, getReviewSchema)
     const reviews = await findReviews(req.user, active, modelId)
+    res.setHeader('x-count', reviews.length)
```
[the remark this answers is not in the document]
Comment (JR40159) [reply backend/src/routes/v2/review/getReviews.ts:30]: http headers are case-insensitive?

File: backend/src/routes.ts [hunk 3]
```diff
@@ -221,8 +221,8 @@ if (config.experimental.v2) {
   server.get('/api/v2/schema/:schemaId', ...getSchemaV2)
   server.post('/api/v2/schemas', ...postSchemaV2)
 
-  server.get('/api/v2/reviews', ...getReviews)
-  server.get('/api/v2/reviews/count', ...getReviewsCount)
+  server.get('/api/v2/reviews/:modelId?', ...getReviews)
+  server.post('/api/v2/reviews/:modelId/:semver/:role', ...postReviewResponse)
```
[the remark this answers is not in the document]
Comment (JR40159) [reply backend/src/routes.ts:225]: See previous comment and I would also add that these pieces of data are used to identify the specific resource so, for me, should be part of the URI. The body should contain the data that the user is wanting to write, not an identifier of that data

File: backend/src/routes/v2/review/getReviews.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply backend/src/routes/v2/review/getReviews.ts:30]: They're not case-sensitive, but I can see us keeping it upper-case to be consistent in other areas of the code but overall I don't think it's a massive issue if it's kept as-is.

[the remark this answers is not in the document]
Comment (JR40159) [reply backend/src/routes/v2/review/getReviews.ts:30]: I also noticed that when unit testing, the header name gets lowercased anyway regardless of what case is used here

**Review by a3957273 — APPROVED** [review]

**Pull request merged**